test.lisp

```diff
@@ -1,22 +1,80 @@
 (in-package :cl)
 
 (defpackage :inferior-shell-test
-  (:use :cl :inferior-shell :hu.dwim.stefil))
+  (:use :cl :inferior-shell :fiveam))
 
 (in-package :inferior-shell-test)
 
 (declaim (optimize (speed 1) (debug 3) (space 3)))
 
-(defsuite* (test-suite
-            :in root-suite
-            :documentation "Testing inferior-shell"))
+(def-suite inferior-shell-test
+  :description "Suite of tests for inferior shell.")
 
-(defun do-test-inferior-shell ()
-  (is (equal (run/ss '(echo (1) "2" (+ 3))) "1 2 3"))
-  (is (equal (run/ss "echo 1    2        3") "1 2 3"))
-  (is (equal (run/ss `(pipe (echo (+ hel "lo,") world)
-                            (tr "hw" "HW") (sed -e "s/$/!/")))
-             "Hello, World!")))
+(def-test basics (:suite inferior-shell-test)
+  (is (equal "1 2 3"
+             (run/ss '(echo (1) "2" (+ 3)))))
+  (is (equal "1 2 3"
+             (run/ss "echo 1    2        3")))
+  (is (equal "Hello, World!"
+             (run/ss `(pipe (echo (+ hel "lo,") world)
+                            (tr "hw" "HW") (sed -e "s/$/!/")))))
+  (is (equal "hello"
+             (run/ss `(pipe (echo ,(format nil "hello~%world"))
+                            (grep "h"))))))
 
-(deftest test-inferior-shell ()
-  (do-test-inferior-shell))
+(def-test tokens (:suite inferior-shell-test)
+  (is (equal "Hello"
+             (run/ss `(echo "Hello"))))
+  (is (equal "hello"
+             (run/ss `(echo hello))))
+  (is (equal "hello"
+             (run/ss `(echo |Hello|))))
+  (is (equal " hello "
+             (run/ss `(echo | Hello |))))
+  (is (equal "--hello"
+             (run/ss `(echo :hello)))))
+
+(def-test + (:suite inferior-shell-test)
+  (is (equal (format nil "Hello world!")
+             (run/ss `(echo (+ "Hello " world!))))))
+
+(def-test and (:suite inferior-shell-test)
+  (is (equal (format nil "hello~%world")
+             (run/ss `(and (echo hello)
+                           (echo world)
+                           (return 1)
+                           (echo good morning))
+                     :on-error nil))))
+
+(def-test or (:suite inferior-shell-test)
+  (is (equal "hello"
+             (run/ss `(or (return 1)
+                          (echo hello)
+                          (echo world))))))
+
+(def-test fork (:suite inferior-shell-test)
+  (uiop:with-temporary-file (:pathname file)
+    (run/ss `(echo (> ,file)))
+    (run/ss `(fork (sleep "0.2") (echo a (> ,file))))
+    (is (equal "" (run/ss `(cat ,file))))
+    (sleep 0.5)
+    (is (equal "a" (run/ss `(cat ,file))))))
```

Comment:
I admit I don't like `sleep` in tests: it slows down testing and will fail under load. Can you instead talk through named pipes to synchronize? Or at least add a TODO to that effect?